### PR TITLE
Ported inheritance test projects from CppWinRT

### PR DESCRIPTION
**Pull request by ujjwalchadha**

### CODE REVIEW

**Comment by kennykerr** [comment]
Need to figure out why the build is failing.

File: test_component_base/HierarchyA.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace winrt::test_component_base::implementation
+{
+    HierarchyA::HierarchyA(hstring const& name)
+    {
+        throw hresult_not_implemented();
+    }
+    void HierarchyA::HierarchyA_Method()
+    {
+        //test_component_base::HierarchyA a = *this;
```
Comment (Scottj1s) [inline test_component_base/HierarchyA.cpp:13]: cruft?

File: test_component_base/packages.config [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.Windows.CppWinRT" version="2.0.200729.8" targetFramework="native" />
```
Comment (Scottj1s) [inline test_component_base/packages.config:3]: this is a pretty old version of cppwinrt - should we go with latest?

**Comment by Scottj1s** [comment]
> C:\source\repos\CsWinRT\TestWinRT\test_component_derived\test_component_derived.vcxproj(44,3): error MSB4019: The imported project "C:\source\repos\CsWinRT\cppwinrt.props" was not found. Confirm that the expression in the Import declaration "C:\source\repos\CsWinRT\\cppwinrt.props" is correct, and that the file exists on disk.

should this be checked in?

---
Refers to: test_component_derived/test_component_derived.log:1 in 0978d6f. [](commit_id = 0978d6f2c7fd9a001420da48c8d2935276422cf5, deletion_comment = False)

**Comment by Scottj1s** [comment]
> Need to figure out why the build is failing.

https://github.com/actions/checkout/issues/383

**Review by Scottj1s — APPROVED** [review]

**Pull request merged**